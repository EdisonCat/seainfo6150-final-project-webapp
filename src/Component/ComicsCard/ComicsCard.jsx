import React from "react";
import styles from "./ComicsCard.module.css";
import {useHistory} from "react-router-dom";
export default function ComicsCard(props) {
    const history = useHistory();
    const handleClick = () => {
        history.push({pathname: `/comics/${props.item.id}`});
    }
    return (
        <div className={styles.container}>
            <div className={styles.thumbnail}><img src={props.item.thumbnail.path + "." + props.item.thumbnail.extension} alt="thumbnail" /></div>
            <div className={styles.info}>
                <h1>{props.item.title}</h1>
                <div className={styles.cta}>
                    <button onClick={() => handleClick()}>More Info</button>
                </div>
            </div>
        </div>
    );
}