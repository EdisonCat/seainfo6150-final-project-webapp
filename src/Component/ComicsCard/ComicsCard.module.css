@import url('https://fonts.googleapis.com/css2?family=Work+Sans:wght@300&display=swap');
.container {
    max-width: 100%;
    border: 1px solid white;
    box-shadow: 2px 5px 5px white;
    color: white;
    display: flex;
    border-radius: 50px;
    padding: 22px;
    margin: 20px;
	font-family: 'Work Sans', sans-serif;
}
.thumbnail {
    max-width: 100%;
    height: auto;
    margin-right: 50px;
}
.info {
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: space-between;
}
.cta {
    display: flex;
    align-items: center;
    justify-content: center;
}
.cta button {
    background: white;
    width: 100px;
    transition: 0.5s;
}
.cta button:hover {
    background: red;
    color: white;
    width: 120px;
}
@media screen and (max-width: 800px) {
    .container {
        flex-direction: column;
    }
    .thumbnail {
        margin-right: 0;
    }
}