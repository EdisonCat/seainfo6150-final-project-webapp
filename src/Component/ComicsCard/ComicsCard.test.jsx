import React from "react";
import ComicsCard from "./ComicsCard";
it("rendered ComicsCard correctly", () => {
	const item = {
		id: 0,
		title: "",
		name: "",
		thumbnail: {
			path: "",
			extension: "",
		},
	};
	// eslint-disable-next-line no-undef
	const { container } = render(<ComicsCard item={item} />);
	expect(container).toMatchSnapshot();
});