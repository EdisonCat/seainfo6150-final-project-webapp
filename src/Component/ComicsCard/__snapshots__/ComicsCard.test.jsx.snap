// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered ComicsCard correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="thumbnail"
    >
      <img
        alt="thumbnail"
        src="."
      />
    </div>
    <div
      class="info"
    >
      <h1 />
      <div
        class="cta"
      >
        <button>
          More Info
        </button>
      </div>
    </div>
  </div>
</div>
`;
